config/environment.rb

```diff
@@ -4,3 +4,7 @@
 
 # Initialize the Rails application.
 Rails.application.initialize!
+
+if rails5? && theme_exists?
+  load_theme_assets
```

Comment:
If we call `Rails.application.config.assets.paths.unshift theme_asset_path` before the `initialize!` it's too early and the theme paths are "prepended" to an empty array and get bumped to the back when the main paths are loaded

If we do anything to `Rails.application.config.assets.paths` after the `initialize!`, it's too late and appears to be functioning as a cache at this point - the config options having already been passed to initialize 

The hack asks the theme - using the full path otherwise it will start from the config dir and go wrong - to update `Rails.application.assets.paths` directly as that's the underlying `Sprockets.environment` so we can force the issue by calling `prepend_path`, but probably shouldn't as that's making the theme code do a lot of the heavy lifting.

Reply:
But it does illustrate what's going on so leaving it here for now so we're not relying on uncommitted code on my laptop